rest-assured/src/main/groovy/io/restassured/assertion/DetailedCookieAssertion.groovy

```diff
@@ -35,6 +35,9 @@ class DetailedCookieAssertion {
         Cookie cookie = cookiesInHeader.get(cookieName)
         if (cookie == null) {
             cookie = responseCookies.get(cookieName)
+        } else if (cookie.getExpiryDate() == null) {
+            Cookie cookieFromResponse = responseCookies.get(cookieName);
+            cookie = new Cookie.Builder(cookie).setExpiryDate(cookieFromResponse.getExpiryDate()).build()
```

Comment:
Hi @johanhaleby, did the `CookieTest.can_receive_detailed_cookies` test pass with your change? It would actually need to be `cookie.getExpiryDate() == null`. The previous test started to break after updating `spring.version` in the `spring-mock-mvc` module.

In the previous version defined in `spring.version`, the call `Cookie cookie = cookiesInHeader.get(cookieName)` applied `null` to variable `cookie`, but in this new version of spring, `cookie` has value, but the value of the `expiryDate` variable is null when it comes from `cookiesInHeader`. So to solve this problem what I did was check if `expiryDate` is `null`, and if it is null I update its value directly from `cookieFromResponse`.